ibc-core/ics02-client/src/handler/create_client.rs

```diff
@@ -29,15 +28,7 @@ where
 
     client_state.verify_consensus_state(consensus_state)?;
 
-    let client_type = client_state.client_type();
-
-    let client_id = ClientId::new(client_type, id_counter).map_err(|e| {
-        ClientError::ClientIdentifierConstructor {
```

Comment:
We can remove this [`ClientError::ClientIdentifierConstructor`](https://github.com/cosmos/ibc-rs/blob/eafdd300727ebc310b2189406cd0bc50f94d2c67/ibc-core/ics02-client/types/src/error.rs#L19-L24) variant. Essentially it was more of identifier errors than being client related.